VirtualPetAdoptionCenter.Web/Controllers/PetController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using VirtualPetAdoptionCenter.Core.Services;
+using VirtualPetAdoptionCenter.Models.Account;
+using VirtualPetAdoptionCenter.Models.Enums;
+
+namespace VirtualPetAdoptionCenter.Web.Controllers
+{
+
+    [ApiController]
+    [Route("[controller]")]
+    public class PetController : Controller
+    {
+        private readonly IPetService _petService;
+        private readonly IAchievementService _achievementTypeService;
+        public PetController(IPetService petService, IAchievementService achievementTypeService)
+        {
+            _petService = petService;
+            _achievementTypeService = achievementTypeService;
+        }   
+
+        [HttpPost]
+        [Route(nameof(Adopt))]
+        public IActionResult Adopt([FromForm]int petId)
+        {
+            var userId = HttpContext.Session.GetInt32(Core.Constants.UserCookieKey);
```

Comment:
I think it's better to add a check if User isn't null, or either add [Authorize] below the method.